rocky_worlds_utils/archive_utils.py

```diff
@@ -0,0 +1,150 @@
+from astroquery.mast import MastMissions
+import numpy as np
+
+
+def check_jwst_observations(ra, dec, radius=0.1):
+    """Check MAST to see if target ra and dec has JWST observations.
+
+    Parameters
+    ----------
+    ra : float
+        Target Right Ascension (units: deg)
+    dec : float
+        Target Declination (units: deg)
+    radius : float
+        Search radius around ra and dec [default: 0.1]
+
+    Returns
+    -------
+    results : astropy.Table.table
+        Astropy table of query results
+    """
+    regionCoords = SkyCoord(ra, dec, unit=("deg", "deg"))
+    missions = MastMissions(mission="jwst")
+    results = missions.query_region(
+        regionCoords,
+        radius=radius,
+        select_cols=[
+            "targprop",
+            "targ_ra",
+            "targ_dec",
+            "instrume",
+            "exp_type",
+            "opticalElements",
+            "date_obs",
+            "duration",
+            "program",
+            "observtn",
+            "visit",
+            "pi_name",
+            "proposal_type",
+            "proposal_cycle",
+            "targtype",
+            "access",
+        ],
+    )
+
+    return results
+
+
+def check_jwst_observation_type(
+    target_name, period, planet_ephemeris, jwst_observations
+):
+    """
+    This fough code tries to figure out, given some target information and
+    an observation start and end time,what exoplanet event is being
+    observed: transit, eclipse, phase curve, or nothing.
+
+    This code may be buggy and has not been well-tested.
+
+    WARNING: This only makes any sense if you already know what host star
+    is being observing for the observation start/end, and then test the
+    planets in that system. Otherwise, you can put in any time range,
+    and for a large enough sample of planets, you will probably randomly
+    hit on a transit/eclispe/phase curve event. This may be obvious but
+    there you are.
+
+    Parameters
+    ----------
+    target_name : str
+        Name of your target
+    period : float
+        Period of exoplanet
+    planet_ephemeris : float
+        The reference time for a transit mid-point.
+    jwst_observations : astropy.Table.table
+        Astropy table of results from astroquery.mast.MastMissions('jwst') query.
+        See `query_mast_jwst_archive` function.
+    """
+
+    # Create time objects for API data to get them in same units as Exoplanet Archive data.
+    obs_start = (
+        Time(jwst_observations["date_obs"], format="isot", scale="utc").jd * u.day
+    )
+    obs_end = obs_start + (jwst_observations["duration"] * u.second).to(u.day)
+
+    for row, (start, end) in enumerate(zip(obs_start, obs_end)):
+        n_start = (start - planet_ephemeris) / period
+        n_end = (end - planet_ephemeris) / period
+        n = int(n_start)
+
+        print(
+            f"This observation starts at n={n_start:0.3f} orbits after ephemeris and ends at n={n_end:0.3f} after"
+        )
```

Comment:
can this return a (single) string instead of printing?

Reply:
We can but we are looping over multiple observations in the function and I intend to use it in a notebook.

Reply:
Realized that for the notebook to work, some more work needed to be done on the `check_jwst_event_type`. It now returns a list of the event types as they appear in the input table. Still working out some kinks with this one.

Reply:
Hey @kecnry I have been working on this recently. Maybe if you wanted to take a look at some of my changes and let me know what you think and we can get this in. Thanks!